src/Stratis.Bitcoin.Features.RPC/Controllers/APIFullNodeController.cs

```diff
@@ -0,0 +1,330 @@
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Controllers;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.Features.RPC.Models;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.Utilities.JsonErrors;
+
+namespace Stratis.Bitcoin.Features.RPC.Controllers
+{
+    [Route("api/[controller]")]
+    public class APIFullNodeController : FeatureController
+    {
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        private readonly IPooledTransaction pooledTransaction;
+
+        /// <summary>An interface implementation used to retrieve unspent transactions from a pooled source.</summary>
+        private readonly IPooledGetUnspentTransaction pooledGetUnspentTransaction;
+
+        /// <summary>An interface implementation used to retrieve unspent transactions.</summary>
+        private readonly IGetUnspentTransaction getUnspentTransaction;
+
+        private readonly INetworkDifficulty networkDifficulty;
+
+        /// <summary>Manager of the longest fully validated chain of blocks.</summary>
+        private readonly IConsensusLoop consensusLoop;
+
+        private readonly Network network;
+
+        private readonly ConcurrentChain chain;
+
+        private readonly IChainState chainState;
+
+        private readonly IConnectionManager connectionManager;
+
+        public APIFullNodeController(
+            ILoggerFactory loggerFactory,
+            Network network,
+            ConcurrentChain chain,
+            IChainState chainState,
+            IConnectionManager connectionManager,
+            IPooledTransaction pooledTransaction = null,
+            IPooledGetUnspentTransaction pooledGetUnspentTransaction = null,
+            IGetUnspentTransaction getUnspentTransaction = null,
+            INetworkDifficulty networkDifficulty = null,
+            IConsensusLoop consensusLoop = null,
+            IFullNode fullNode = null,
+            NodeSettings nodeSettings = null)
+            : base(
+                  fullNode: fullNode,
+                  nodeSettings: nodeSettings)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.pooledTransaction = pooledTransaction;
+            this.pooledGetUnspentTransaction = pooledGetUnspentTransaction;
+            this.getUnspentTransaction = getUnspentTransaction;
+            this.networkDifficulty = networkDifficulty;
+            this.consensusLoop = consensusLoop;
+            this.network = network;
+            this.chain = chain;
+            this.chainState = chainState;
+            this.connectionManager = connectionManager;
+        }
+
+        /// <summary>
+        /// Stops the full node.
+        /// API implementation of RPC call.
+        /// </summary>
+        [Route("stop")]
+        [HttpGet]
+        public async Task<IActionResult> Stop()
+        {
+			await SharedRemoteMethods.Stop(this.FullNode);
```

Comment:
Please replace tabs with spaces everywhere in the changed files (1 tab == 4 spaces)

Reply:
@noescape00 Done. Hadn't set up VS Code to do tabs to spaces yet. Sorry!